packages/itwinui-react/src/core/InputGrid/InputGrid.test.tsx

```diff
@@ -54,3 +59,51 @@ it('should take class and style', () => {
   expect(inputContainer).toHaveClass('iui-input-grid my-class');
   expect(inputContainer.style.width).toBe('50px');
 });
+
+it.each`
```

Comment:
Should we also test cases like:
* When user id is given, we should use that instead
* When `<StatusMessage>` is not provided, there should be no description
* `labelId` should also be there when `hasSelect`
* etc.

Reply:
>     * When user id is given, we should use that instead

Added tests for this.  [`b8952de`](https://github.com/iTwin/iTwinUI/pull/1777/commits/b8952dee31b8de3a751dbf9063bef54af20b1bbe)

I also changed the code to handle explicitly-passed `undefined`.

```jsx
<InputGrid>
  <Label id={undefined}>Label</Label>
  <Input id={undefined} />
  <StatusMessage id={undefined}>Hint message</StatusMessage>
</InputGrid>
```

>     * When `<StatusMessage>` is not provided, there should be no description

I don't see why this is necessary. We don't care that there is no description.


>     * `labelId` should also be there when `hasSelect` 
>     * etc.

These are implementation details that should not be tested. We only care about the final result, i.e. accName and accDesc are present.

Reply:
> > `labelId` should also be there when `hasSelect`
> > etc.
> 
> These are implementation details that should not be tested. We only care about the final result, i.e. accName and accDesc are present.

I mean a test to confirm that id is not added to `<Label>` unnecessarily. Instead, it is only added when there is a `<Select>`.

Reply:
It still doesn't matter, it's an implementation detail. We do not care if the Label has an id, and the developer using this component certainly does not care either.

I recommend reading this article: https://kentcdodds.com/blog/testing-implementation-details

Reply:
I read that article, but I am not suggesting testing implementation details, like the value of `hasSelect`. I am suggesting having a test to make sure we don't add props when they are not needed.

Even though developers may not care whether an id is added or not, I feel adding it unnecessarily is still incorrect, and thus should be tested that it is not happening.

Basically, since you have written logic to prevent this `id` from being added to `Label` when no `Select` is present, I feel this case/logic should be tested (independent of the implementation details needed to implement this case/logic).

Reply:
As discussed via chat, we can skip this test for the reasons you mentioned in this thread 👍